tools/autograd/derivatives.yaml

```diff
@@ -1092,21 +1098,27 @@
 
 - name: reciprocal(Tensor self) -> Tensor
   self: -grad * (result * result).conj()
+  result: auto_element_wise
 
 - name: remainder.Scalar(Tensor self, Scalar other) -> Tensor
   self: grad
+  result: auto_element_wise
 
 - name: remainder.Tensor(Tensor self, Tensor other) -> Tensor
   self: grad
+  # Use addition with zeros to get the proper broadcasting behavior
+  result: self_t + at::zeros_like(other)
```

Comment:
Can you use result? `self_t.expand_as(result)`.

Otherwise, there are ways to compute the broadcast shape like infer_size: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ExpandUtils.h#L14

Reply:
The problem with expand is that it needs to have the same number of dimensions.
But yes, let me check infer_size!

Reply:
> The problem with expand is that it needs to have the same number of dimensions.

Does it?
```
[ins] In [1]: import torch
x = torch.
[ins] In [2]: x = torch.randn(2, 1)

[ins] In [3]: y = torch.randn(3, 1, 4)

[ins] In [4]: result = x + y

[ins] In [5]: x.expand_as(result)
Out[5]:
tensor([[[ 0.7218,  0.7218,  0.7218,  0.7218],
         [-1.7359, -1.7359, -1.7359, -1.7359]],

        [[ 0.7218,  0.7218,  0.7218,  0.7218],
         [-1.7359, -1.7359, -1.7359, -1.7359]],

        [[ 0.7218,  0.7218,  0.7218,  0.7218],
         [-1.7359, -1.7359, -1.7359, -1.7359]]])
```

Reply:
What? Since when is expand doing broadcasting?? :p